File: src/1769.minimum-number-of-operations-to-move-all-balls-to-each-box.1895/solution.ts
function minOperations (boxes: string): number[] {
    const result:number[] = new Array(boxes.length);
    let leftDis = 0;
    let leftCount = 0;
    for (let i = 0; i < boxes.length; i++) {
        result[i] = leftDis;
        if (boxes[i] === '1') {
            leftCount++;
        }
        leftDis += leftCount;
    }
    let rightDis = 0;
    let rightCount = 0;
    for (let i = boxes.length - 1; i > -1; i--) {
        result[i] += rightDis;
        if (boxes[i] === '1') {
            rightCount++;
        }
        rightDis += rightCount;
    }
    return result;
}

File: src/1769.minimum-number-of-operations-to-move-all-balls-to-each-box.1895/solution.java
class Solution {
    public int[] minOperations(String boxes) {
        char[] chars = boxes.toCharArray();
        int[] result = new int[chars.length];
        int count = 0;
        int dis = 0;
        for(int i=0;i< chars.length;i++){
            result[i] = dis;
            if(chars[i] == '1'){
                count++;
            }
            dis += count;
        }
        count = 0;
        dis = 0;
        for(int i=chars.length-1;i>-1;i--){
            result[i] += dis;
            if(chars[i] == '1'){
                count++;
            }
            dis += count;
        }
        
        return result;
    }
}